CMakeLists.txt

```diff
@@ -69,88 +41,36 @@ add_library(roboteam_world_srcs
         src/kalman/kalmanBall.cpp
         src/kalman/kalmanUs.cpp
         src/kalman/kalmanThem.cpp
-
-    # src/world
         src/filtered_world/filtered_world.cpp
-        src/dummy_world/world_dummy.cpp
-
-    )
-
-add_dependencies(roboteam_world_srcs
-    ${${PROJECT_NAME}_EXPORTED_TARGETS}
-    ${catkin_EXPORTED_TARGETS}
-    )
-target_link_libraries(roboteam_world_srcs 
-    ${catkin_LIBRARIES}
-    )
-
-#################
-## Dummy world ##
-#################
+        src/dummy_world/world_dummy.cpp)
 
-add_executable(dummy_world
-        src/dummy_world/dummy_world_main.cpp
-    )
-target_link_libraries(dummy_world
-    roboteam_world_srcs
-    ${catkin_LIBRARIES} ${ARMADILLO_LIBRARIES}
-    )
-add_dependencies(dummy_world
-    ${${PROJECT_NAME}_EXPORTED_TARGETS}
-    ${catkin_EXPORTED_TARGETS}
-    )
 
-####################
-## Filtered world ##
-####################
-
-add_executable(filtered_world
-        src/filtered_world/filtered_world_main.cpp
-    )
-target_link_libraries(filtered_world
-    roboteam_world_srcs
-    ${catkin_LIBRARIES} ${ARMADILLO_LIBRARIES}
+target_include_directories(kalman_world PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:include>
+        PRIVATE src )
 
-        )
-add_dependencies(filtered_world 
-    ${${PROJECT_NAME}_EXPORTED_TARGETS} 
-    ${catkin_EXPORTED_TARGETS}
-    )
+target_link_libraries(kalman_world ${ARMADILLO_LIBRARIES} net roboteam_utils roboteam_proto)
```

Comment:
You should most definitely make sure to link armadillo net and utils privately and proto publicly.